packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1563,6 +1563,34 @@ export const InvalidContentEntryDataError = {
 	hint: 'See https://docs.astro.build/en/guides/content-collections/ for more information on content schemas.',
 } satisfies ErrorData;
 
+/**
+ * @docs
+ * @message
+ * **Example error message:**<br/>
+ * The content loader for the collection **blog** returned an invalid result:<br/>
+ * Content entry `id` must be a string:<br/>
+ * {<br/>
+ * 	 "id": 1,<br/>
+ * 	 "title": "Hello, World!"<br/>
+ * }
+ * @description
+ * A content loader returned an invalid result.
```

Comment:
```suggestion
 * A content loader returned an invalid `id`.
```
Like the title of your PR, if the reason this entry is invalid is only and exactly because of the ID, would it be more helpful to specify this here?

Reply:
I've tried keeping the error as agnostic as possible so it can be used for other validation that might be needed in the future. The actual error content does include the following line:
"Property `id` has to be a string"
And right below that it shows the entry that caused this error.

Reply:
Just checking on this because I'm totally not sure whether this is how we've handled error messages in the past! I'm not sure we've had one that "means different things" and gets a different message based on exactly what went wrong.  Usually it's one particular error, but the dynamic part of the message is e.g. identifying the file name or something like that.

Would love for someone to confirm this! Maybe @ematipico would be in a good position to judge this.

Reply:
Ah fair enough, didn't know that. Gonna wait for Ema's input then

Reply:
If possible, we usually want to create specific messages. These objects are also created in our documentation, so users who land on our docs want to know **exactly** what the error is and how to fix it. 

I share the sentiment of reusing shared "code"/"logic", but in this case we need to bend our vision towards DX and provide specific messages to users.

Reply:
In which case I'll change it to that, thanks a lot for your input!